sklearn/utils/validation.py

```diff
@@ -982,3 +982,121 @@ def check_scalar(x, name, target_type, min_val=None, max_val=None):
 
     if max_val is not None and x > max_val:
         raise ValueError('`{}`= {}, must be <= {}.'.format(name, x, max_val))
+
+
+def check_kernel_eigenvalues(lambdas, warn_on_zeros=False):
+    """Checks that the provided array of kernel eigenvalues for numerical or
```

Comment:
Checks kernel eigenvalues for conditioning issues

is sufficient. The rest can be expanded on below

Reply:
Ok.